qdev_wrappers/dataset/doNd.py

```diff
@@ -1,27 +1,90 @@
-from typing import Callable, Sequence, Union, Tuple, List, Optional
+from contextlib import contextmanager
+from typing import Callable, Sequence, Union, Tuple, List, Optional, Iterator
 import os
 import time
 
 import numpy as np
 import matplotlib
 import matplotlib.pyplot as plt
 
-from qcodes.dataset.measurements import Measurement
+from qcodes.dataset.measurements import Measurement, res_type, DataSaver
 from qcodes.instrument.base import _BaseParameter
 from qcodes.dataset.plotting import plot_by_id
 from qcodes import config
 
+ActionsT = Sequence[Callable[[], None]]
+
+ParamMeasT = Union[_BaseParameter, Callable[[], None]]
+
 AxesTuple = Tuple[matplotlib.axes.Axes, matplotlib.colorbar.Colorbar]
 AxesTupleList = Tuple[List[matplotlib.axes.Axes],
                       List[Optional[matplotlib.colorbar.Colorbar]]]
 AxesTupleListWithRunId = Tuple[int, List[matplotlib.axes.Axes],
                       List[Optional[matplotlib.colorbar.Colorbar]]]
-number = Union[float, int]
 
 
-def do0d(*param_meas:  Union[_BaseParameter, Callable[[], None]],
-         write_period: Optional[float] = None,
-         do_plot: bool = True) -> AxesTupleListWithRunId:
+def _process_params_meas(param_meas: ParamMeasT) -> List[res_type]:
+    output = []
+    for parameter in param_meas:
+        if isinstance(parameter, _BaseParameter):
+            output.append((parameter, parameter.get()))
+        elif callable(parameter):
+            parameter()
+    return output
+
+
+def _register_parameters(
+        meas: Measurement,
+        param_meas: List[ParamMeasT],
+        setpoints: Optional[List[_BaseParameter]] = None
+) -> None:
+    for parameter in param_meas:
+        if isinstance(parameter, _BaseParameter):
+            meas.register_parameter(parameter,
```

Comment:
@Dominik-Vogel is it possible that we can handle array parameters separately to ensure they get saved as blobs in the database. This was the main reason this PR https://github.com/qdev-dk/qdev-wrappers/pull/166 exists.

As is the doNd functions are completely unusable when doing anything with an arrayparameter (the overhead for saving is way too large).

Reply:
Good point! Thank you for notifying us!

Reply:
We will consider these in  a new pr, to keep this one pure refactoring.

Reply:
@GateBuilder can you work on a test for this?